great_expectations/expectations/expectation.py

```diff
@@ -1546,6 +1536,7 @@ def _get_maturity_checklist(
                 tests
             )
         )
+        experimental_checks.append(ExpectationDiagnostics._check_linting(self))
```

Comment:
Should `check_linting` be a public method? I feel little iffy invoking a private method if we can avoid it.

Reply:
All the staticmethod/classmethod decorated methods in the ExpectationDiagnostics class start with `_check_`